src/MoonSharp.Interpreter/Options/FuzzySymbolMatchingBehaviour.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+
+namespace MoonSharp.Interpreter {
+
+	/// <summary>
+	/// A flag that controls if/how symbols (method, property, userdata) are fuzzily matched when they do not exist. Flags can be combined for multiple checks.
+	/// </summary>
+	[Flags]
+	public enum FuzzySymbolMatchingBehaviour {
+
+		/// <summary>No fuzzy matching is performed.</summary>
+		None = 0,
+
+		/// <summary>The first letter of a symbol will be uppercased (to check for common C# naming conventions). For example, testMethod() becomes TestMethod()</summary>
+		UpperFirstLeter = 1,
+
+		/// <summary>Underscores in symbols are converted to camelcase. For example, test_method() becomes testMethod()</summary>
+		Camelify = 2,
+
+		/// <summary>
+		/// Combines both <see cref="UpperFirstLeter"/> and <see cref="Camelify"/>. For example, test_Method_two() becomes TestMethodTwo()
+		/// </summary>
+		All = UpperFirstLeter | Camelify
```

Comment:
_**This is a discussion note, please feel free to argue/disagree**_ 

**I originally wrote this for your first change, but you fixed part of it to use it more like flags -- but the discussion point on ALL still sits here**

I understand what you were trying to do here, but I think this causes some logic problems.

Right now your code will do: 
1. Perform UpperFirstLetter
2. Perform Camelify
3. Perform CamelifyUpperFirstLetter (PascalCase)

In code though you are saying "All" which is only UpperFirstLetter and Camelify (according to the enum) but really means "PascalCase". In your version of the code it does PascalCase & UpperFirstLetter & Camelify inclusively, but won't allow an exclusive PascalCase run.

Also, if we later add other fuzzy methods too here, then all will become a bit more confusing in my opinion. For example, I just got a request for kebab-case and I want to add it so I change the enum:
```
KebabCase = 4
All = UpperFirstLetter | Camelify | Kebab-Case
```
You can see where this starts to get messy, because now if I just wanted to independently run:

1. UpperFirstLetter
2. Camelify
3. Kebab-Case

I automatically do a merge-run on all three as well, because the inclusive All flag.

In other words (and based on current code):
For example, PascalCase is NOT functionally equivalent to UpperFirstLetter | Camelify as those are independent; and there is no ability to run PascalCase with a fall-through on just Camelify while excluding just UpperFirstLetter.

```
None = 0
UpperFirstLetter = 1
Camelify = 2
PascalCase  = 4  // Or CamelifyUpperFirst
```

We don't really need an all as the _default behavior in code_ would be to run them all (exclusively) anyways if the flag is set and we use an &. If we want a mash-up we should define it like PascalCase. Or KebabCaseUpperFirst. That would allow us to easily use mixed-sets of evaluations in the case of fuzzy-matching gone wild.

Just playgrounding it, feel free to edit for discussion: https://dotnetfiddle.net/TP7OOX

Reply:
I understand what you're saying. Originally it was labeled "Both" which didn't make sense, I guess "All" doesn't make any sense either (it defeats my purpose of adding more behavior in the future). So, remove All, add PascalCase, and set the default in ScriptGlobalOptions to UpperFirstLetter | Camelify | PascalCase. I believe this will address your concerns?

Reply:
That should do it!